src/box/space_cache.c

```diff
@@ -71,63 +71,6 @@ space_by_name(const char *name)
 	return (struct space *)mh_strnptr_node(spaces_by_name, space)->val;
 }
 
-/**
- * Visit all spaces and apply 'func'.
- */
-int
-space_foreach(int (*func)(struct space *sp, void *udata), void *udata)
```

Comment:
Why did you move this function? Looks like it's more about the space cache than about a single space.

Reply:
My thoughts were as follows: `space_foreach()` is more about spaces in general rather than space cache. So after I removed direct access to `struct mh_i32ptr_t *spaces`, it can be placed anywhere.